### ISSUE TITLE
ClusterSingletonManager is not getting member UP - cluster is unresponsive

### ISSUE THREAD

**Issue by elyrank** [akka/akka#18519]
we are using 2.3.13 and started getting this new behavior:

restarting nodes one after the other , the ClusterSingletonManager did not get the add(member) command and its member list is empty - does not contain even himself (I saw it in remote debug)
in the logs I'm expecting to see the cluster listener to get member state UP, but that never happens.

the result is that every message designated to sharded actors is not handled and the cluster is as good as dead

logs:

node1 log:

```
2015-09-18 11:20:37,230 INFO  [tor.default-dispatcher-16] [Cluster(akka://ActorSystem)             ]   Cluster Node [akka.tcp://ActorSystem@node002.test.com:2553] - Welcome from [akka.tcp://ActorSystem@node001.test.com:2553]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-2, akkaTimestamp=08:20:37.230UTC, akkaSource=Cluster(akka://ActorSystem)
2015-09-18 11:20:37,539 INFO  [ctor.default-dispatcher-2] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Start -> Younger]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-3, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/ClusterListener, akkaTimestamp=08:20:37.539UTC
2015-09-18 11:20:37,539 INFO  [ctor.default-dispatcher-2] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Start -> Younger]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-2, akkaTimestamp=08:20:37.539UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/ConversationControllerCoordinator
2015-09-18 11:20:37,539 INFO  [ctor.default-dispatcher-2] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Start -> Younger]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-17, akkaTimestamp=08:20:37.539UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/DialogControllerCoordinator
2015-09-18 11:21:07,340 INFO  [ctor.default-dispatcher-3] [a.c.pattern.ClusterSingletonManager     ]   Ignoring TakeOver request in [Younger] from [akka.tcp://ActorSystem@node001.test.com:2553].   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-3, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/DialogControllerCoordinator, akkaTimestamp=08:21:07.339UTC
2015-09-18 11:21:07,341 INFO  [tor.default-dispatcher-17] [a.c.pattern.ClusterSingletonManager     ]   Ignoring TakeOver request in [Younger] from [akka.tcp://ActorSystem@node001.test.com:2553].   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-17, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/ClusterListener, akkaTimestamp=08:21:07.340UTC
2015-09-18 11:21:07,341 INFO  [tor.default-dispatcher-17] [a.c.pattern.ClusterSingletonManager     ]   Ignoring TakeOver request in [Younger] from [akka.tcp://ActorSystem@node001.test.com:2553].   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-17, akkaTimestamp=08:21:07.341UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/ConversationControllerCoordinator
2015-09-18 11:21:07,357 INFO  [ctor.default-dispatcher-5] [a.c.pattern.ClusterSingletonManager     ]   Younger observed OldestChanged: [Some(akka.tcp://ActorSystem@node001.test.com:2553) -> myself]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-16, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/ConversationControllerCoordinator, akkaTimestamp=08:21:07.355UTC
2015-09-18 11:21:07,358 INFO  [ctor.default-dispatcher-5] [a.c.pattern.ClusterSingletonManager     ]   Younger observed OldestChanged: [Some(akka.tcp://ActorSystem@node001.test.com:2553) -> myself]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-2, akkaTimestamp=08:21:07.356UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/DialogControllerCoordinator
2015-09-18 11:21:07,358 INFO  [ctor.default-dispatcher-5] [a.c.pattern.ClusterSingletonManager     ]   Younger observed OldestChanged: [Some(akka.tcp://ActorSystem@node001.test.com:2553) -> myself]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-6, akkaTimestamp=08:21:07.357UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/ClusterListener
2015-09-18 11:21:07,363 INFO  [tor.default-dispatcher-16] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Younger -> BecomingOldest]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-16, akkaTimestamp=08:21:07.362UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/ConversationControllerCoordinator
2015-09-18 11:21:07,368 INFO  [ctor.default-dispatcher-2] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Younger -> BecomingOldest]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-6, akkaTimestamp=08:21:07.368UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/ClusterListener
2015-09-18 11:21:07,368 INFO  [ctor.default-dispatcher-2] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Younger -> BecomingOldest]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-2, akkaTimestamp=08:21:07.368UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/DialogControllerCoordinator
2015-09-18 11:21:07,389 INFO  [ctor.default-dispatcher-4] [a.c.pattern.ClusterSingletonManager     ]   Hand-over in progress at [akka.tcp://ActorSystem@node001.test.com:2553]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-6, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/DialogControllerCoordinator, akkaTimestamp=08:21:07.389UTC
2015-09-18 11:21:07,391 INFO  [ctor.default-dispatcher-4] [a.c.pattern.ClusterSingletonManager     ]   Hand-over in progress at [akka.tcp://ActorSystem@node001.test.com:2553]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-3, akkaTimestamp=08:21:07.391UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/ConversationControllerCoordinator
2015-09-18 11:21:07,392 INFO  [ctor.default-dispatcher-6] [a.c.pattern.ClusterSingletonManager     ]   Hand-over in progress at [akka.tcp://ActorSystem@node001.test.com:2553]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-3, akkaTimestamp=08:21:07.392UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/ClusterListener
2015-09-18 11:21:08,033 INFO  [ctor.default-dispatcher-3] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.contrib.pattern.ShardCoordinator$Internal$Register] from Actor[akka://ActorSystem/user/sharding/DialogController#1953252199] to Actor[akka://ActorSystem/deadLetters] was not delivered. [1] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-6, akkaTimestamp=08:21:08.033UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:08,042 INFO  [ctor.default-dispatcher-6] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.contrib.pattern.ShardCoordinator$Internal$Register] from Actor[akka://ActorSystem/user/sharding/ConversationController#1443067327] to Actor[akka://ActorSystem/deadLetters] was not delivered. [2] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-6, akkaTimestamp=08:21:08.041UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:08,127 INFO  [ctor.default-dispatcher-2] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.cluster.GossipStatus] from Actor[akka://ActorSystem/system/cluster/core/daemon#-49199166] to Actor[akka://ActorSystem/deadLetters] was not delivered. [3] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-17, akkaTimestamp=08:21:08.127UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:08,362 INFO  [tor.default-dispatcher-17] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.cluster.ClusterHeartbeatSender$Heartbeat] from Actor[akka://ActorSystem/system/cluster/core/daemon/heartbeatSender#-637253714] to Actor[akka://ActorSystem/deadLetters] was not delivered. [4] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-17, akkaTimestamp=08:21:08.362UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:09,122 INFO  [ctor.default-dispatcher-2] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.cluster.GossipStatus] from Actor[akka://ActorSystem/system/cluster/core/daemon#-49199166] to Actor[akka://ActorSystem/deadLetters] was not delivered. [5] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-2, akkaTimestamp=08:21:09.122UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:09,362 INFO  [ctor.default-dispatcher-2] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.cluster.ClusterHeartbeatSender$Heartbeat] from Actor[akka://ActorSystem/system/cluster/core/daemon/heartbeatSender#-637253714] to Actor[akka://ActorSystem/deadLetters] was not delivered. [6] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-2, akkaTimestamp=08:21:09.362UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:10,032 INFO  [ctor.default-dispatcher-4] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.contrib.pattern.ShardCoordinator$Internal$Register] from Actor[akka://ActorSystem/user/sharding/DialogController#1953252199] to Actor[akka://ActorSystem/deadLetters] was not delivered. [7] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-4, akkaTimestamp=08:21:10.032UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:10,042 INFO  [ctor.default-dispatcher-4] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.contrib.pattern.ShardCoordinator$Internal$Register] from Actor[akka://ActorSystem/user/sharding/ConversationController#1443067327] to Actor[akka://ActorSystem/deadLetters] was not delivered. [8] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-4, akkaTimestamp=08:21:10.042UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:10,122 INFO  [ctor.default-dispatcher-4] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.cluster.GossipStatus] from Actor[akka://ActorSystem/system/cluster/core/daemon#-49199166] to Actor[akka://ActorSystem/deadLetters] was not delivered. [9] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-4, akkaTimestamp=08:21:10.122UTC, akkaSource=akka://ActorSystem/deadLetters
2015-09-18 11:21:10,362 INFO  [tor.default-dispatcher-18] [ctorRefProvider$RemoteDeadLetterActorRef]   Message [akka.cluster.ClusterHeartbeatSender$Heartbeat] from Actor[akka://ActorSystem/system/cluster/core/daemon/heartbeatSender#-637253714] to Actor[akka://ActorSystem/deadLetters] was not delivered. [10] dead letters encountered, no more dead letters will be logged. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-4, akkaSource=akka://ActorSystem/deadLetters, akkaTimestamp=08:21:10.362UTC
2015-09-18 11:21:13,138 WARN  [ctor.default-dispatcher-4] [akka.remote.ReliableDeliverySupervisor  ]   Association with remote system [akka.tcp://ActorSystem@node001.test.com:2553] has failed, address is now gated for [5000] ms. Reason: [Association failed with [akka.tcp://ActorSystem@node001.test.com:2553]] Caused by: [Connection refused: node001.test.com/192.168.22.19:2553]   |TAGS|sourceThread=ActorSystem-akka.remote.default-remote-dispatcher-8, akkaTimestamp=08:21:13.137UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/system/endpointManager/reliableEndpointWriter-akka.tcp%3A%2F%2FActorSystem%40node001.test.com%3A2553-2
2015-09-18 11:21:18,366 WARN  [tor.default-dispatcher-18] [akka.remote.ReliableDeliverySupervisor  ]   Association with remote system [akka.tcp://ActorSystem@node001.test.com:2553] has failed, address is now gated for [5000] ms. Reason: [Association failed with [akka.tcp://ActorSystem@node001.test.com:2553]] Caused by: [Connection refused: node001.test.com/192.168.22.19:2553]   |TAGS|sourceThread=ActorSystem-akka.remote.default-remote-dispatcher-7, akkaTimestamp=08:21:18.366UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/system/endpointManager/reliableEndpointWriter-akka.tcp%3A%2F%2FActorSystem%40node001.test.com%3A2553-2
2015-09-18 11:21:24,036 WARN  [ctor.default-dispatcher-3] [akka.remote.ReliableDeliverySupervisor  ]   Association with remote system [akka.tcp://ActorSystem@node001.test.com:2553] has failed, address is now gated for [5000] ms. Reason: [Association failed with [akka.tcp://ActorSystem@node001.test.com:2553]] Caused by: [Connection refused: node001.test.com/192.168.22.19:2553]   |TAGS|sourceThread=ActorSystem-akka.remote.default-remote-dispatcher-8, akkaTimestamp=08:21:24.036UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/system/endpointManager/reliableEndpointWriter-akka.tcp%3A%2F%2FActorSystem%40node001.test.com%3A2553-2
2015-09-18 11:21:29,126 WARN  [tor.default-dispatcher-17] [akka.remote.ReliableDeliverySupervisor  ]   Association with remote system [akka.tcp://ActorSystem@node001.test.com:2553] has failed, address is now gated for [5000] ms. Reason: [Association failed with [akka.tcp://ActorSystem@node001.test.com:2553]] Caused by: [Connection refused: node001.test.com/192.168.22.19:2553]   |TAGS|sourceThread=ActorSystem-akka.remote.default-remote-dispatcher-7, akkaTimestamp=08:21:29.126UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/system/endpointManager/reliableEndpointWriter-akka.tcp%3A%2F%2FActorSystem%40node001.test.com%3A2553-2
2015-09-18 11:21:34,127 INFO  [ctor.default-dispatcher-4] [Cluster(akka://ActorSystem)             ]   Cluster Node [akka.tcp://ActorSystem@node002.test.com:2553] - Marking exiting node(s) as UNREACHABLE [Member(address = akka.tcp://ActorSystem@node001.test.com:2553, status = Exiting)]. This is expected and they will be removed.   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-3, akkaTimestamp=08:21:34.126UTC, akkaSource=Cluster(akka://ActorSystem)
2015-09-18 11:21:34,144 INFO  [tor.default-dispatcher-18] [Cluster(akka://ActorSystem)             ]   Cluster Node [akka.tcp://ActorSystem@node002.test.com:2553] - Leader is removing exiting node [akka.tcp://ActorSystem@node001.test.com:2553]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-3, akkaTimestamp=08:21:34.144UTC, akkaSource=Cluster(akka://ActorSystem)
2015-09-18 11:21:34,146 INFO  [tor.default-dispatcher-18] [a.c.pattern.ClusterSingletonManager     ]   Previous oldest [akka.tcp://ActorSystem@node001.test.com:2553] removed   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-18, akkaTimestamp=08:21:34.146UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/DialogControllerCoordinator
2015-09-18 11:21:34,148 INFO  [ctor.default-dispatcher-5] [a.c.pattern.ClusterSingletonManager     ]   Previous oldest [akka.tcp://ActorSystem@node001.test.com:2553] removed   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-5, akkaTimestamp=08:21:34.148UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/ClusterListener
2015-09-18 11:21:34,148 INFO  [ctor.default-dispatcher-5] [a.c.pattern.ClusterSingletonManager     ]   Previous oldest [akka.tcp://ActorSystem@node001.test.com:2553] removed   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-5, akkaTimestamp=08:21:34.148UTC, akkaSource=akka.tcp://ActorSystem@node002.test.com:2553/user/sharding/ConversationControllerCoordinator
2015-09-18 11:21:42,898 INFO  [tor.default-dispatcher-17] [Cluster(akka://ActorSystem)             ]   Cluster Node [akka.tcp://ActorSystem@node002.test.com:2553] - Node [akka.tcp://ActorSystem@node001.test.com:2553] is JOINING, roles []   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-18, akkaTimestamp=08:21:42.898UTC, akkaSource=Cluster(akka://ActorSystem)
2015-09-18 11:21:43,133 INFO  [ctor.default-dispatcher-3] [Cluster(akka://ActorSystem)             ]   Cluster Node [akka.tcp://ActorSystem@node002.test.com:2553] - Leader is moving node [akka.tcp://ActorSystem@node001.test.com:2553] to [Up]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-5, akkaTimestamp=08:21:43.133UTC, akkaSource=Cluster(akka://ActorSystem)
```

node2 log:
```
2015-09-18 11:21:44,070 INFO  [ctor.default-dispatcher-4] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Start -> Younger]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-18, akkaSource=akka.tcp://ActorSystem@node001.test.com:2553/user/ClusterListener, akkaTimestamp=08:21:44.069UTC
2015-09-18 11:21:44,086 INFO  [tor.default-dispatcher-17] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Start -> Younger]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-17, akkaSource=akka.tcp://ActorSystem@node001.test.com:2553/user/sharding/DialogControllerCoordinator, akkaTimestamp=08:21:44.086UTC
2015-09-18 11:21:44,211 INFO  [ctor.default-dispatcher-5] [a.c.pattern.ClusterSingletonManager     ]   ClusterSingletonManager state change [Start -> Younger]   |TAGS|sourceThread=ActorSystem-akka.actor.default-dispatcher-5, akkaTimestamp=08:21:44.211UTC, akkaSource=akka.tcp://ActorSystem@node001.test.com:2553/user/sharding/ConversationControllerCoordinator
```

**Comment by elyrank** [comment]
some more details - we are using auto down = off
static seed nodes
using cluster sharding extension

we are also triggering a leave request when the node is shutting down , and waiting to get member removed before shutting down.

**Comment by patriknw** [comment]
@elyrank Thanks for the logs. I can see exactly what is wrong, and it has actually been fixed in master (2.4.0-RC3) and Typesafe Reactive Platform (in case you are a subscriber).

It's a missing `gotoOldest` at this line: https://github.com/akka/akka/blob/v2.3.13/akka-contrib/src/main/scala/akka/contrib/pattern/ClusterSingletonManager.scala#L485

here is corresponding line in master: https://github.com/akka/akka/blob/master/akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala#L534

**Comment by elyrank** [comment]
can you add it to 2.3.15 ?
